src/main/java/com/rultor/agents/github/CommentsTag.java

```diff
@@ -94,44 +106,191 @@ public CommentsTag(final Github ghub) {
         final String tag = req.xpath("args/arg[@name='tag']/text()").get(0);
         final Releases.Smart rels = new Releases.Smart(issue.repo().releases());
         final URI home = new Home(xml).uri();
+        final List<DefaultArtifactVersion> previous = this.versions(rels);
+        boolean release = true;
         if (rels.exists(tag)) {
-            final Release.Smart rel = new Release.Smart(rels.find(tag));
-            rel.body(
-                String.format(
-                    "%s\n\nSee also #%d and [build log](%s)",
-                    rel.body(), issue.number(), home
-                )
-            );
-            issue.comments().post(
-                String.format(
-                    CommentsTag.PHRASES.getString("CommentsTag.duplicate"),
-                    tag
-                )
-            );
-            Logger.info(this, "duplicate tag %s commented", tag);
-        } else {
-            final Repo repo = issue.repo();
-            final Date prev = CommentsTag.previous(repo);
-            final Release.Smart rel = new Release.Smart(
-                rels.create(tag.trim())
-            );
-            rel.name(issue.title());
-            rel.prerelease(true);
-            rel.body(
-                String.format(
-                    // @checkstyle LineLength (1 line)
-                    "See #%d, release log:\n\n%s\n\nReleased by Rultor %s, see [build log](%s)",
-                    issue.number(),
-                    new CommitsLog(repo).build(prev, rel.publishedAt()),
-                    Manifests.read("Rultor-Version"), home
-                )
-            );
-            Logger.info(this, "tag %s created and commented", tag);
+            this.logDuplicateTag(issue, tag, rels, home);
+            release = false;
+        }
+        if (release && !CommentsTag.valid(tag)) {
+            this.logInvalidTag(issue, tag);
+            release = false;
+        }
+        if (release && !CommentsTag.valid(tag, previous)) {
+            this.logTagTooOld(issue, tag, previous);
+            release = false;
+        }
+        if (release) {
+            this.release(issue, tag, rels, home);
         }
         return new Directives();
     }
 
     /**
+     * Create the release and add a comment.
+     * @param issue The issue to add a comment to.
+     * @param tag The tag for the release.
+     * @param rels The existing releases.
+     * @param home The URI of home.
+     * @throws IOException If there is a problem.
+     * @checkstyle ParameterNumberCheck (15 lines)
```

Comment:
@gumbelmj instead of suppressing the check that is supposed to indicate to you that your method is too complex by using this many inputs, you need to create a method in compliance with the check.